assets/src/components/shuttleMapPage.tsx

```diff
@@ -57,39 +74,161 @@ const ShuttleMapPage = (): ReactElement<HTMLDivElement> => {
-          shapes={shapes}
-          trainVehicles={trainVehicles}
-          onPrimaryVehicleSelect={(vehicle) => dispatch(selectVehicle(vehicle))}
-          selectionKey={followerResetKey}
-          tileType={tileType}
-        >
-          <LayersControl.WithTileContext
-            setTileType={(tileType: TileType) =>
-              dispatch(setTileType("shuttleMap", tileType))
-            }
-          />
-        </MapFollowingSelectionKey>
-      </div>
+  const setShouldFollow = (controller: FollowerController) => {
+    setFollowerController(controller)
+    setFollowActive(controller !== false ? true : false)
+  }
+
+  const [userLocation, setUserLocation] =
+    useState<null | GeolocationCoordinates>(null)
+
+  const followPositions = useMemo(
+    () =>
+      followerController
+        ? {
+            "user-location": userLocation
+              ? [latLng(userLocation.latitude, userLocation.longitude)]
+              : [],
+            "vehicle-location": vehiclePositions,
+          }[followerController]
+        : [],
+    [followerController, userLocation, vehiclePositions]
+  )
+
+  useEffect(() => {
+    if (followerController === false) {
+      setShouldFollow("vehicle-location")
+    }
+    // This effect is an "event", and not "reactive" to all it's dependencies.
+    // Because we _only_ want to run this when the run id's change, we need to
+    // suppress the exhaustive-deps lint until `useEffectEvent` is stabilized
+    //
+    // See the react docs:
+    // > ## Is it okay to suppress the dependency linter instead?
+    // > After `useEffectEvent` becomes a stable part of React,
+    // > we recommend never suppressing the linter.
+    // > -- https://react.dev/learn/separating-events-from-effects#is-it-okay-to-suppress-the-dependency-linter-instead
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedShuttleRunIds])
+
+  const followerFn = usePickerContainerFollowerFn()
+
+  return (
+    <div className="c-shuttle-map__map">
+      <Map
+        stateClasses={FollowerStatusClasses(
+          followerController == "vehicle-location"
+        )}
+        vehicles={selectedShuttles}
+        selectedVehicleId={selectedVehicleId}
+        shapes={selectedRouteShapes}
+        trainVehicles={trainVehicles}
+        onPrimaryVehicleSelect={selectVehicle}
+        tileType={tileType}
+      >
+        <LayersControl.WithTileContext setTileType={setTileType} />
+
+        <UserLocationControl
+          title="Show your current location"
+          onClick={() => {
+            setShouldFollow("user-location")
+          }}
+          disabled={followerController == "user-location"}
+        />
+        <RecenterControl
+          position="topright"
+          recenter={() => setShouldFollow("vehicle-location")}
+        />
+
+        <>
+          {(followerController === "user-location" || userLocation) && (
+            <UserLocation onLocationUpdate={(l) => setUserLocation(l)} />
```

Comment:
```suggestion
            <UserLocation onLocationUpdate={setUserLocation} />
```

I think you can just pass `setUserLocation` in rather than wrap it in an anonymous function.

Reply:
Oh right, I had a `console.debug` statement here for a bit and hadn't cleaned this up. Thanks!